### PR TITLE
Update redis_exporter.pp

### PR DESCRIPTION
**Pull request by allangood**
Add support to "-redis.file" instead of specify all hosts with "$addr" array variable.

#### Pull Request (PR) description

This change will bring more flexibility to specify hosts, passwords and alias to each redis instance/server. Also, with this change the password will not be exposed to all users through "ps".

#### This Pull Request (PR) fixes the following issues
* Do not exposes server information to all users
* Add capability to specify many different servers/instances
* Add deprecation alert to the $addr variable

### CODE REVIEW

File: manifests/redis_exporter.pp [hunk 1]
```diff
@@ -82,15 +90,17 @@
...
   String $download_url_base,
-  Array[String] $extra_groups,
   String $group,
+  Array[String] $extra_groups,
   String $package_ensure,
   String $package_name,
   String $user,
   String $version,
+  String $redis_exporter_config_file,
+  Array[String] $addr,
```
Comment (bastelfreak) [inline manifests/redis_exporter.pp:102]: is  there a reason for reordering the existing parameters?

**Comment by bastelfreak** [comment]
Hi @allangood, thanks for the work. Can you take a look at the failing travis jobs?

File: manifests/redis_exporter.pp (hunk 1 above)
Comment (allangood) [reply manifests/redis_exporter.pp:102]: Hi @bastelfreak , the only reason is to make things easy to read grouping variables by its type.

**Comment by allangood** [comment]
Trevis didn't like that "redis_exporter_config_file" doesn't have a default value.
I tried to use the same standard for the others exporters like "snmp_exporter" and "blackbox_exporter" that uses a "config_file" variable without a default value.
I'm going to fix it.
Sorry to make many PR for this little change.
Regards.

**Pull request closed without merging**